### PR TITLE
Retry by next tick for execution poll error

### PR DESCRIPTION
**Pull request by tnsetting**
# Hey, I just made a Pull Request!

## Description
Retry by next tick for execution poll error. Do nothing when execution polling failed as in real life, there should always be an execution in Flyte. We don't use the grpc retries because it would block the thread for a while. 
## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/FlyteRunnerHandler.java [hunk 1]
```diff
@@ -109,6 +109,9 @@ private void pollingExecution(RunState state, EventRouter eventRouter) {
 
     try {
       flyteRunner.poll(getExecutionId(state), state);
+    } catch (FlyteRunner.PollingException e) {
+      final var errMessage = "Could not poll execution for " + state.workflowInstance() + " , will be tried";
```
Comment (ckiosidis) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/FlyteRunnerHandler.java:113]: you mean retried here?

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java [hunk 2]
```diff
@@ -201,25 +201,28 @@ private String getCause(RunState runState) {
...
-      // TODO: handle gRPC application level retry or do not fail but let ticking thread in PersistentStateManager retry
-      throw new PollingException(flyteExecutionId, e);
-    } catch (Exception e) {
-      // TODO: do not fail and let ticking thread in PersistentStateManager retry
-      throw new PollingException(flyteExecutionId, e);
+
+      return;
+    }
+
+    if (execution != null) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java:224]: This guard is not needed because we will not reach here if exception happens.

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/NoopFlyteRunner.java [hunk 3]
```diff
@@ -47,13 +47,12 @@ public void terminateExecution(RunState runState, FlyteExecutionId flyteExecutio
   }
 
   @Override
-  public void poll(final FlyteExecutionId flyteExecutionId, final RunState runState)
-      throws PollingException {
-    throw new PollingException("Cannot poll for execution: " + flyteExecutionId.toUrn());
+  public void poll(final FlyteExecutionId flyteExecutionId, final RunState runState) {
+    throw new IllegalStateException("Cannot poll execution : " + flyteExecutionId);
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/NoopFlyteRunner.java:51]: Do we need this exception? It breaks the contract.

File: styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java [hunk 4]
```diff
@@ -281,7 +280,6 @@ public void testCreateNoopFlyteRunner() {
     assertThat(flyteRunner, notNullValue());
     assertThat(flyteRunner.isEnabled(), is(false));
     assertThrows(FlyteRunner.CreateExecutionException.class, () -> flyteRunner.createExecution(null, null, null, null));
-    assertThrows(FlyteRunner.PollingException.class, () -> flyteRunner.poll(FlyteExecutionId.create("flyte-test","testing","test"), null));
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java:284]: We can still test this by calling the method without asserting anything.

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java [hunk 5]
```diff
@@ -376,19 +376,17 @@ public void testPollingExceptionFlyteAdminClientExecutionNotFound() {
     doThrow(new StatusRuntimeException(Status.NOT_FOUND))
         .when(flyteAdminClient).getExecution(anyString(), anyString(), anyString());
 
-    assertThrows(
-        FlyteRunner.ExecutionNotFoundException.class,
-        () -> flyteRunner.poll(FLYTE_EXECUTION_ID, RUN_STATE));
+    flyteRunner.poll(FLYTE_EXECUTION_ID, RUN_STATE);
+    verify(stateManager).receiveIgnoreClosed(
+        Event.runError(WORKFLOW_INSTANCE, "Could not find execution: " + FLYTE_EXECUTION_ID.toUrn()),-1);
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:381]:
```suggestion
        Event.runError(WORKFLOW_INSTANCE, "Could not find execution: " + FLYTE_EXECUTION_ID.toUrn()), -1);
```

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java [hunk 6]
```diff
@@ -397,7 +395,7 @@ public void testPollingException() {
         .when(flyteAdminClient).getExecution(anyString(), anyString(), anyString());
 
     assertThrows(
-        FlyteRunner.PollingException.class,
+        RuntimeException.class,
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:398]: This test case is weird. Do we really expect this is a case? This is like testing any arbitrary Java method. :)

File: styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java (hunk 4 above)
Comment (honnix) [reply styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java:284]: We are a bit over testing here because as long as we can create an instance of `NoopFlyteRunner` and `isEnabled` is `false`, we are good. Testing other behaviour of the class belongs to the unit test of its own.

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/NoopFlyteRunnerTest.java [hunk 7]
```diff
@@ -64,7 +64,7 @@ public void testTerminateExecutionThrowsException() {
   @Test
   public void testPollingThrowsException() {
     assertThrows(
-        FlyteRunner.PollingException.class,
+        IllegalStateException.class,
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/NoopFlyteRunnerTest.java:67]: As commented above, we can test it doesn't throw.

**Review by honnix — COMMENTED** [review]
Some minor comments, other looks good.

File: styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java (hunk 4 above)
Comment (tnsetting) [reply styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java:284]: Shall I remove the `CreateExecution` as well?

Comment (honnix) [reply styx-scheduler-service/src/test/java/com/spotify/styx/StyxSchedulerTest.java:284]: I think we only need to test the instance is of type NoopFlyteRunner, which covers non-null check as well. isEnabled is always false.

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/NoopFlyteRunner.java [hunk 8]
```diff
@@ -27,7 +27,7 @@
 /**
  * No-op {@code FlyteRunner} meant to be used when Flyte is disabled or not available.
  */
-class NoopFlyteRunner implements FlyteRunner {
+public class NoopFlyteRunner implements FlyteRunner {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/NoopFlyteRunner.java:30]: nit: we can mark this as visible for test.

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/NoopFlyteRunnerTest.java [hunk 9]
```diff
@@ -63,10 +63,7 @@ public void testTerminateExecutionThrowsException() {
 
   @Test
   public void testPollingThrowsException() {
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/NoopFlyteRunnerTest.java:65]: testPolling?

**Review by honnix — APPROVED** [review]
Small issues. Feel free to merge after addressing those.
